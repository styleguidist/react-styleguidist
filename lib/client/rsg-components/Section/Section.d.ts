import React from 'react';
import * as Rsg from '../../../typings';
declare const Section: React.FunctionComponent<{
    section: Rsg.Section;
    depth: number;
}>;
export default Section;
