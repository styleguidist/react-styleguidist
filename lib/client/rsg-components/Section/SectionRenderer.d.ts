import React from 'react';
import { JssInjectedProps } from 'rsg-components/Styled';
interface SectionRendererProps extends JssInjectedProps {
    slug: string;
    depth: number;
    name?: string;
    description?: string;
    content?: React.ReactNode;
    components?: React.ReactNode;
    sections?: React.ReactNode;
    isolated?: boolean;
    pagePerSection?: boolean;
    [prop: string]: any;
}
export declare const SectionRenderer: React.FunctionComponent<SectionRendererProps>;
declare const _default: React.ComponentType<Pick<SectionRendererProps, string | number>>;
export default _default;
