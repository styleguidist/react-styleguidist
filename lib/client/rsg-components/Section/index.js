export { default } from 'rsg-components/Section/Section';
export * from 'rsg-components/Section/Section';