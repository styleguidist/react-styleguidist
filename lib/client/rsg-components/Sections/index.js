export { default } from 'rsg-components/Sections/Sections';