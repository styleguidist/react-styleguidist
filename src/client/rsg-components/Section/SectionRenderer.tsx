import React from 'react';
import PropTypes from 'prop-types';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import SectionHeading from 'rsg-components/SectionHeading';
import Markdown from 'rsg-components/Markdown';
import * as Rsg from '../../../typings';

const styles = ({ space }: Rsg.Theme) => ({
	root: {
		marginBottom: space[4],
	},
});

interface SectionRendererProps extends JssInjectedProps {
	slug: string;
	depth: number;
	name?: string;
	description?: string;
	content?: React.ReactNode;
	components?: React.ReactNode;
	sections?: React.ReactNode;
	isolated?: boolean;
	pagePerSection?: boolean;
	[prop: string]: any;
}

export const SectionRenderer: React.FunctionComponent<SectionRendererProps> = (allProps) => {
	const {
		classes,
		name,
		slug,
		content,
		components,
		sections,
		depth,
		description,
		pagePerSection,
	} = allProps;

	return (
		<section className={classes.root} data-testid={`section-${slug}`}>
			{name && (
				<SectionHeading
					depth={depth}
					id={slug}
					slotName="sectionToolbar"
					pagePerSection={pagePerSection}
					slotProps={allProps}
				>
					{name}
				</SectionHeading>
			)}
			{description && <Markdown text={description} />}
			{content}
			{sections}
			{components}
		</section>
	);
};

SectionRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	name: PropTypes.string,
	description: PropTypes.string,
	slug: PropTypes.string.isRequired,
	content: PropTypes.any,
	components: PropTypes.any,
	sections: PropTypes.any,
	isolated: PropTypes.bool,
	depth: PropTypes.number.isRequired,
	pagePerSection: PropTypes.bool,
};

export default Styled<SectionRendererProps>(styles)(SectionRenderer);
