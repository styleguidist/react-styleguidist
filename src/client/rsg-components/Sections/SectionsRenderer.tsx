import React from 'react';
import PropTypes from 'prop-types';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';

const styles = () => ({
	// Just default jss-isolate rules
	root: {},
});

interface SectionsRendererProps extends JssInjectedProps {
	children: React.ReactNode;
}

export const SectionsRenderer: React.FunctionComponent<SectionsRendererProps> = ({
	classes,
	children,
}) => {
	return <section className={classes.root}>{children}</section>;
};

SectionsRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	children: PropTypes.any,
};

export default Styled<SectionsRendererProps>(styles)(SectionsRenderer);
