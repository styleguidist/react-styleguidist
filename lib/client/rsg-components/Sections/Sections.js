import "core-js/modules/es.array.filter";
import "core-js/modules/es.array.map";
import React from 'react';
import PropTypes from 'prop-types';
import Section from 'rsg-components/Section';
import SectionsRenderer from 'rsg-components/Sections/SectionsRenderer';

var Sections = function Sections(_ref) {
  var sections = _ref.sections,
      depth = _ref.depth;
  return /*#__PURE__*/React.createElement(SectionsRenderer, null, sections.filter(function (section) {
    return !section.externalLink;
  }).map(function (section, idx) {
    return /*#__PURE__*/React.createElement(Section, {
      key: idx,
      section: section,
      depth: depth
    });
  }));
};

Sections.propTypes = {
  sections: PropTypes.array.isRequired,
  depth: PropTypes.number.isRequired,
  root: PropTypes.bool
};
export default Sections;