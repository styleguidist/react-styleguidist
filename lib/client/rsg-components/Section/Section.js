import "core-js/modules/es.symbol";
import "core-js/modules/es.symbol.description";
import "core-js/modules/es.function.name";
import React from 'react';
import PropTypes from 'prop-types';
import Examples from 'rsg-components/Examples';
import Components from 'rsg-components/Components';
import Sections from 'rsg-components/Sections';
import SectionRenderer from 'rsg-components/Section/SectionRenderer';
import { useStyleGuideContext } from 'rsg-components/Context';
import { DisplayModes } from '../../consts';

var Section = function Section(_ref) {
  var section = _ref.section,
      depth = _ref.depth;

  var _useStyleGuideContext = useStyleGuideContext(),
      displayMode = _useStyleGuideContext.displayMode,
      pagePerSection = _useStyleGuideContext.config.pagePerSection;

  var name = section.name,
      slug = section.slug,
      filepath = section.filepath,
      content = section.content,
      components = section.components,
      sections = section.sections,
      description = section.description,
      exampleMode = section.exampleMode,
      usageMode = section.usageMode;
  var contentJsx = Array.isArray(content) ? /*#__PURE__*/React.createElement(Examples, {
    examples: content,
    name: name,
    exampleMode: exampleMode
  }) : null;
  var componentsJsx = components && /*#__PURE__*/React.createElement(Components, {
    usageMode: usageMode,
    exampleMode: exampleMode,
    components: components,
    depth: depth + 1
  });
  var sectionsJsx = sections && /*#__PURE__*/React.createElement(Sections, {
    sections: sections,
    depth: depth + 1
  });
  return /*#__PURE__*/React.createElement(SectionRenderer, {
    description: description,
    pagePerSection: pagePerSection,
    name: name,
    slug: slug,
    filepath: filepath,
    content: contentJsx,
    components: componentsJsx,
    sections: sectionsJsx,
    isolated: displayMode !== DisplayModes.all,
    depth: depth
  });
};

Section.propTypes = {
  section: PropTypes.any.isRequired,
  depth: PropTypes.number.isRequired
};
export default Section;