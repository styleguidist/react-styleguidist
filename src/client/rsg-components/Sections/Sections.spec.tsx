import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import noop from 'lodash/noop';
import Section from '../Section';
import Sections from './Sections';
import StyledSectionsRenderer, { SectionsRenderer } from './SectionsRenderer';

const sections = [
	{
		name: 'Foo',
		content: [
			{
				type: 'code',
				content: '<button>OK</button>',
				evalInContext: noop,
			},
		],
		components: [],
	},
	{
		name: 'Bar',
		content: [
			{
				type: 'markdown',
				content: 'Hello *world*!',
			},
		],
		components: [],
	},
	{
		sections: [
			{
				name: 'One',
				content: [],
			},
			{
				name: 'Two',
				content: [],
			},
		],
	},
] as any;

it('should render component renderer', () => {
	const renderer = createRenderer();
	renderer.render(<Sections sections={sections} depth={3} />);

	expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('render should render styled component', () => {
	const renderer = createRenderer();
	renderer.render(
		<StyledSectionsRenderer>
			<Section key={0} section={sections[0]} depth={3} />
			<Section key={1} section={sections[1]} depth={3} />
			<Section key={2} section={sections[2]} depth={3} />
		</StyledSectionsRenderer>
	);

	expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('render should render component', () => {
	const renderer = createRenderer();
	renderer.render(
		<SectionsRenderer classes={{}}>
			<Section key={0} section={sections[0]} depth={3} />
			<Section key={1} section={sections[1]} depth={3} />
			<Section key={2} section={sections[2]} depth={3} />
		</SectionsRenderer>
	);

	expect(renderer.getRenderOutput()).toMatchSnapshot();
});
