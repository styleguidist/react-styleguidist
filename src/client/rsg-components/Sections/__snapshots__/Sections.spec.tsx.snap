// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render component 1`] = `
<section>
  <Section
    depth={3}
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
        ],
        "name": "Foo",
      }
    }
  />
  <Section
    depth={3}
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ],
        "name": "Bar",
      }
    }
  />
  <Section
    depth={3}
    section={
      Object {
        "sections": Array [
          Object {
            "content": Array [],
            "name": "One",
          },
          Object {
            "content": Array [],
            "name": "Two",
          },
        ],
      }
    }
  />
</section>
`;

exports[`render should render styled component 1`] = `
<SectionsRenderer
  classes={
    Object {
      "root": "rsg--root-0",
    }
  }
>
  <Section
    depth={3}
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
        ],
        "name": "Foo",
      }
    }
  />
  <Section
    depth={3}
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ],
        "name": "Bar",
      }
    }
  />
  <Section
    depth={3}
    section={
      Object {
        "sections": Array [
          Object {
            "content": Array [],
            "name": "One",
          },
          Object {
            "content": Array [],
            "name": "Two",
          },
        ],
      }
    }
  />
</SectionsRenderer>
`;

exports[`should render component renderer 1`] = `
<Styled(Sections)>
  <Section
    depth={3}
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
        ],
        "name": "Foo",
      }
    }
  />
  <Section
    depth={3}
    section={
      Object {
        "components": Array [],
        "content": Array [
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ],
        "name": "Bar",
      }
    }
  />
  <Section
    depth={3}
    section={
      Object {
        "sections": Array [
          Object {
            "content": Array [],
            "name": "One",
          },
          Object {
            "content": Array [],
            "name": "Two",
          },
        ],
      }
    }
  />
</Styled(Sections)>
`;
